File: solutions/cpp/pangram/1/pangram.cpp
#include "pangram.h"

namespace pangram {

bool is_pangram(std::string sentence)
{
    bool is_used['z' - 'a'];
    for (char& c : sentence) {
        if (('a' <= c && c <= 'z') || ('A' <= c && c <= 'Z'))
            is_used[tolower(c) - 'a'] = true;
    }

    for (bool used : is_used)
        if (!used)
            return false;
    return true;
}

}  // namespace pangram

File: solutions/zig/pangram/1/pangram.zig
const std = @import("std");

pub fn isPangram(str: []const u8) bool {
    var letters: @Vector(26, bool) = [_]bool{false} ** 26;
    for (str) |x| {
        if (std.ascii.isAlphabetic(x)) {
            letters[std.ascii.toLower(x) - 'a'] = true;
        }
    }
    return @reduce(.And, letters);
}
